packages/teams/src/share-or-unshare-items-to-team.ts

```diff
@@ -0,0 +1,47 @@
+import { ArcGISRequestError, IRequestOptions } from "@esri/arcgis-rest-request";
+import { IItem } from "@esri/arcgis-rest-types";
+import { shareItemToGroups, unshareItemFromGroups } from "@esri/hub-common";
+
+/**
+ * Process an array of items, either sharing them to the group
+ * or unsharing them. Defaults to sharing items. Returns an object comprised of
+ * array of successes, and failures/errors.
+ *
+ * @export
+ * @param {string} groupId | group we are sharing/unsharing to
+ * @param {IItem[]} items | array of items we are sharing/unsharing
+ * @param {IRequestOptions} requestOptions | Auth
+ * @param {("share" | "unshare")} [action="share"] | share or unshare, defaults to share
+ * @return {*}
+ */
+export async function shareOrUnshareItemsToTeam(
```

Comment:
This method isn't really specific to `teams`, so would probably be better in `hub-common` if we need to add this.

More importantly, I'm not sold on the idea of having a single method handle both the sharing and unsharing of multiple items. I would be more supportive of adding separate `shareItemsToGroups` and `unshareItemFromGroups` methods, although I _think_ the REST API already supports this; we just need a pair of REST.js methods to facilitate this. You can see this in action by:
1. Sign into AGOL
2. Click `Groups` in top nav
3. Filter to groups your auth'd user owns
4. Click `Add items to group`
5. Tick checkbox for multiple items
6. Click `Add items` button
7. Observe the following request and response network tab:

**CURL for request:**
```curl
curl 'https://qa-pre-a-hub.mapsqa.arcgis.com/sharing/rest/content/users/rweber_5_qa_pre_a_hub/shareItems' \
  -H 'authority: qa-pre-a-hub.mapsqa.arcgis.com' \
  -H 'pragma: no-cache' \
  -H 'cache-control: no-cache' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="96", "Google Chrome";v="96"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.55 Safari/537.36' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'content-type: application/x-www-form-urlencoded' \
  -H 'accept: */*' \
  -H 'origin: https://qa-pre-a-hub.mapsqa.arcgis.com' \
  -H 'sec-fetch-site: same-origin' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-dest: empty' \
  -H 'referer: https://qa-pre-a-hub.mapsqa.arcgis.com/home/group.html?id=6c71237b6d66401c9c1ed99defc68b6a' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'cookie: ago_shield_token=home_X6h7ruI8opPKl; esri_aopc=a0ULjoiuyOwKP6XUzFaRdqA..jp_H3c3d_IxNXPXWB3LWcxWKWoTFX84vptQkGteyd92KtrJ_v4nXpmkoceYJhUymChtzWMq69n-4Ausw7PsI1epO2SE7aEEn54cQVrYJ-higb5I3zMg7p3WB7hBatJ0KvM2Y7BLU4Ha8K3eIA-n9iHvJJkhPfnH1D-PXYYB8E7PEgSCt4Y2jlvTV_Nf478wRh0GaRXVZDiEowcokEmzDRe_xJTW1_F2eox95Qmn0P0qJyzLalOp5JUOeFofTO3lpkbQUKrtf; esri_locale=en-us; APSW-query=%7B%22expires%22%3A1639080397836%7D; ESRI_Content=%7B%22folderId%22%3A%22%2F%22%2C%22isAll%22%3Afalse%2C%22isHome%22%3Atrue%7D' \
  --data-raw 'f=json&items=381ff797cc554e66adce841bdee3b0ea%2C610ff40e017a4ad581337a7b1a9238e0&groups=6c71237b6d66401c9c1ed99defc68b6a&confirmItemControl=true&token=g_zOd4IeNIk6Ucy-lwxzNNYLdzmYWLQ4eCvm_5NoGsUG35goPBQymHu4kA7TGaAM3YQvyweUJu1jkM4JTTsU_keucl1sBOeoaEKTS5nKH8NE13sSBZdrm-fL5sdzPNHe0uyIFLC8jCDewX_k_R5-B0hZ9WL27FHyaTJ_Ansvp0b1KIAFGVBTkqBNve3Je3rMDf556IhQJkDyG2O9HfeF5sM9p3hRSvT4goxGuPAPk5s.' \
  --compressed
```

**Response:**
```json
{
	"results": [{
		"itemId": "610ff40e017a4ad581337a7b1a9238e0",
		"success": true,
		"notSharedWith": []
	}, {
		"itemId": "381ff797cc554e66adce841bdee3b0ea",
		"success": true,
		"notSharedWith": []
	}]
}
```